src/screens/WalletConnect/Provider.js

```diff
@@ -0,0 +1,280 @@
+/* @flow */
+import React, { useState, useEffect } from "react";
+import WalletConnect from "@walletconnect/client";
+import { useSelector } from "react-redux";
+import { parseCallRequest } from "@ledgerhq/live-common/lib/walletconnect";
+import { saveWCSession, getWCSession } from "../../db";
+import { accountScreenSelector } from "../../reducers/accounts";
+import { NavigatorName, ScreenName } from "../../const";
+import { navigate, navigationRef } from "../../rootnavigation";
+
+export const context = React.createContext();
+
+export const STATUS = {
+  DISCONNECTED: 0x00,
+  CONNECTING: 0x01,
+  ERROR: 0x02,
+  CONNECTED: 0x03,
+};
+
+const Provider = ({ children }: { children: React$Node }) => {
+  const [session, setSession] = useState({});
+  const [connector, setConnector] = useState();
+  const [status, setStatus] = useState(STATUS.DISCONNECTED);
+  const [error, setError] = useState();
+  const [initDone, setInitDone] = useState(false);
+  const [currentCallRequestId, setCurrentCallRequestId] = useState();
+  const [currentCallRequestError, setCurrentCallRequestError] = useState();
+  const [currentCallRequestResult, setCurrentCallRequestResult] = useState();
+  const [dappInfo, setDappInfo] = useState();
+  const [approveSession, setApproveSession] = useState();
+
+  const connect = ({ uri, account }) => {
+    setStatus(STATUS.CONNECTING);
+    setError();
+
+    let connector;
+
+    try {
+      connector = new WalletConnect(
+        session.session
+          ? {
+              session: session.session,
+            }
+          : {
+              // Required
+              uri,
+              // Required
+              clientMeta: {
+                description: "LedgerLive CLI",
+                url: "https://ledger.fr",
+                icons: [
+                  "https://avatars0.githubusercontent.com/u/9784193?s=400&v=4",
```

Comment:
probably an icon of Ledger Live is more appropriated.

Reply:
I will update, do I use this play store link https://play-lh.googleusercontent.com/RVKjd96rcTjiAnr45Gy6Nj2kCJ4_opdU2mrop7KftfyRhPWJf5ukvUR_Gi9AtOA920I or should I use something else ? @gre